lib/vsc/accounting/data/parser.py

```diff
@@ -0,0 +1,180 @@
+#
+# https://github.com/sisc-hpc/vsc-accounting-brussel
+#
+# vsc-accounting-brussel is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation v2.
+#
+# vsc-accounting-brussel is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with vsc-manage.  If not, see <http://www.gnu.org/licenses/>.
+#
+##
+"""
+Parser of data files for vsc.accounting
+
+@author: Alex Domingo (Vrije Universiteit Brussel)
+"""
+
+import os
+import appdirs
+import pkg_resources
+import json
+
+from bs4 import BeautifulSoup
+
+from vsc.utils import fancylogger
+from vsc.accounting.filetools import make_dir, copy_file
+from vsc.accounting.exit import error_exit
+
+
+class DataFile:
+    """
+    Parse contents of a data file in user's data dir
+    Copy data file to user's data dir if it does not exist
+    Supported formats: JSON, HTML
+    """
+
+    def __init__(self, datafile, mandatory=True):
+        """
+        Set the location of the data file, copy it if it does not exist and read its contents
+        - datafile: (string) name of the data file or full path to data file
+        - mandatory: (boolean) mandatory files must already exist in user's data dir or be copied from package
+        """
+        self.log = fancylogger.getLogger(name=self.__class__.__name__)
+
+        datafile = os.path.expanduser(datafile)
+
+        if os.path.isabs(datafile):
+            # Directly read from absolute path
+            self.datafile = {'path': datafile}
+            # By default try to read file from absolute path
+            self.datafile.update({'readable': True})
+        else:
+            # Use datafile in user data directory
+            self.datafile = {
+                'name': datafile,
+                'dir': appdirs.user_data_dir('vsc-accounting'),
+            }
+            self.datafile.update({'path': os.path.join(self.datafile['dir'], self.datafile['name'])})
+
+            # Copy data file from package contents (if it is missing in user's data dir)
+            # Failed copies are only fatal for mandatory data files
+            try:
+                self.copy_pkgdata()
+            except FileNotFoundError as err:
+                self.datafile.update({'readable': False})
+                if mandatory:
+                    error_exit(self.log, err)
+            else:
+                self.datafile.update({'readable': True})
+
+        # Read contents of data file
+        if self.datafile['readable']:
+            try:
+                self.read_data()
+            except ValueError as err:
+                error_exit(self.log, err)
+
+    def copy_pkgdata(self, force=False):
```

Comment:
this json data in the package is bogus in any case, so we actually have to force the user to input something?

Reply:
The templates and styles for the reports in HTML are suitable to be edited by the user. Therefore, it is more convenient to have those copied to the user's home.

Regarding the default definition of nodegroups, I updated it in https://github.com/sisc-hpc/vsc-accounting-brussel/pull/1/commits/c3e60be133d22ebf36986e70ad5b89775f1131a8 to make it somewhat usable without further configuration. Now it defines a single group of nodes matching any hostnames. Nonetheless, this file will also be copied to user's home because it can/may be edited by the user.

On the other hand, the path to the nodegroup file defined in the `specsheet` configuration option already accepts full paths.